helm/cluster-api-provider-aws/templates/eks-bootstrap-components.yaml

```
@@ -0,0 +1,266 @@
+  resources:
+  - eksconfigs/status
+  verbs:
+  - get
+  - patch
+  - update
+- apiGroups:
+  - cluster.x-k8s.io
+  resources:
+  - clusters
+  - machinepools
+  - machines
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - controlplane.cluster.x-k8s.io
+  resources:
+  - awsmanagedcontrolplanes
+  verbs:
+  - get
+  - list
+  - watch
+- apiGroups:
+  - exp.cluster.x-k8s.io
+  resources:
+  - machinepools
+  verbs:
+  - get
+  - list
+  - watch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+  name: capa-eks-bootstrap-proxy-role
+rules:
+- apiGroups:
+  - authentication.k8s.io
+  resources:
+  - tokenreviews
+  verbs:
+  - create
+- apiGroups:
+  - authorization.k8s.io
+  resources:
+  - subjectaccessreviews
+  verbs:
+  - create
+---
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+  name: capa-eks-bootstrap-metrics-reader
+rules:
+- nonResourceURLs:
+  - /metrics
+  verbs:
+  - get
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+  name: capa-eks-bootstrap-leader-election-rolebinding
+  namespace: capa-eks-bootstrap-system
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: capa-eks-bootstrap-leader-election-role
+subjects:
+- kind: ServiceAccount
+  name: capa-eks-bootstrap-controller-manager
+  namespace: capa-eks-bootstrap-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+  name: capa-eks-bootstrap-manager-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: capa-eks-bootstrap-manager-role
+subjects:
+- kind: ServiceAccount
+  name: capa-eks-bootstrap-controller-manager
+  namespace: capa-eks-bootstrap-system
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+  name: capa-eks-bootstrap-proxy-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: capa-eks-bootstrap-proxy-role
+subjects:
+- kind: ServiceAccount
+  name: capa-eks-bootstrap-controller-manager
+  namespace: capa-eks-bootstrap-system
+---
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
+  name: capa-eks-bootstrap-controller-manager-metrics-service
+  namespace: capa-eks-bootstrap-system
+spec:
+  ports:
+  - name: https
+    port: 8443
+    targetPort: https
+  selector:
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    cluster.x-k8s.io/provider: bootstrap-eks
+    control-plane: controller-manager
+  name: capa-eks-bootstrap-controller-manager
```

```suggestion
  name: {{ include "resource.default.name" . }}
```
Naming can be used like this - it works a bit easier.

---

Easiest to check my PR maybe? Uncertain